master/buildbot/wamp/connector.py

```diff
@@ -0,0 +1,140 @@
+# This file is part of .  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright  Team Members
+
+from autobahn.twisted.wamp import ApplicationSession
+from autobahn.twisted.wamp import Service
+from autobahn.wamp.exception import TransportLost
+
+from twisted.internet import defer
+from twisted.python import failure
+from twisted.python import log
+
+from buildbot.util import service
+
+
+class MasterService(ApplicationSession, service.AsyncMultiService):
```

Comment:
BuildbotService is really for Services configured in the master.cfg. Here this is a internal service.

AsyncMultiService allows to manage the list of childrens, and lifecycle capability to the ApplicationSession.

We could separate it in attributes, but I think this would just make the thing more complicated.